### ISSUE TITLE
FEATURE REQUEST: Airwave Simulator (keep fan running for X minutes after thermostatOperatingState goes to IDLE)

### ISSUE THREAD

**Issue by johnny2678** [tonesto7/nest-manager#472]
Correct me if I'm mistaken, but looks like as soon as the cooling setpoint is reached, the thermostat shuts the compressor and the fan down.

Is there a way to tap into the Nest Airwave functionality? Seems like Nest does some monitoring of your system to determine how much cooling it can do by cutting off the compressor early and just running the fan over the coils.

[https://nest.com/support/article/What-is-Airwave](https://nest.com/support/article/What-is-Airwave)

Even if they don't expose that learning data - this should be easy enough to simulate.   

Consider the following scenario:
```
Cooling Setpoint (existing parameter): 77 degrees
Cooling Threshold (existing parameter): 1.5 degrees
```
**Current behavior**: When the temperature in the house gets to 78.5, the AC kicks in and runs until the temperature in the house falls below 77.

Same scenario as above with additional proposed airwave settings:
```
Cooling Setpoint (existing parameter): 77 degrees
Cooling Threshold (existing parameter): 1.5 degrees
Airwave Threshold (proposed parameter): 0.25 degrees
Airwave (fan) Run Time (proposed parameter): 8 Minutes
```

**Proposed behavior with Airwave enabled**: When the temperature in the house gets to 78.5, the AC kicks in and runs until the temperature in the house reaches 77.25 degrees (Cooling setpoint + Airwave Threshold).  Then the `thermostatOperatingState` switches to `idle` while the fan stays on for an additional 8 minutes.

If you really wanted to be informative, you could include an airwave run report in the logs:

ex `fan state back to auto: airwave ran for 8 minutes and lowered the temperature in the house by an additional 0.3 degrees F with the compressor off`

This would allow the user to adjust the Airwave Threshold/Airwave fan run time to reach their target temps.

Caveat: As proposed, this would only work for those with multiple (>3?) alternate temp sensors, so the average has a potential to create a fractional value OR sensors that are capable of high precision readings.  The ds18b20 sensors I'm using are precise down to a 16th of an degree.

Thoughts?  Feel free to call me crazy, just think it would be one more amazing feature to this already amazing app.

edit: formatting

**Comment by imnotbob** [comment]
We should move this discussion to a private message in the forum.  Please reach out to me there

**Comment by johnny2678** [comment]
Sure, no problem.  Are you Anthony (@tonesto7) or Eric (@E_Sch)?  All I know is you're not Bob ;)

**Comment by imnotbob** [comment]
Eric

**Comment by brandiniman** [comment]
Running the fan after the compressor shuts off will raise the indoor humidity level and lower indoor comfort. Putting a lockout to only operate below a certain threshold would be a good idea- but I think Nest does this already which is why it most likely doesn't do this for you now. Mine doesn't use airwave likely due to my home being at 50-55% RH most times. Turning the compressor and fan off at the same time lets the indoor coil condense water to the drain and dry between cycles moving minimal humidity to the home.

**Comment by johnny2678** [comment]
Good point on setting a Humidity threshold so simulated airwave only kicks off under a set percentage.  I have a whole home dehumidifier so internal humidity usually stays between 40-50%.

You sound like you know what you are talking about.  Is the airwave stuff just marketing hype or is there something to it?  Makes sense to me to blow the fan across the coils for another 2-3 minutes after cooling, but what do I know ;)

**Comment by brandiniman** [comment]
It's completely a good idea to run indoor air that's warmer than the coil over the coil that's still cooler- but only if the indoor humidity doesn't rise too much while running fan-only. Eventually the coil and moisture on it will warm and evaporate into the indoor air, which makes the job of cooling the house harder as cooling 'wet' air is much harder than cooling dry air.

**Comment by johnny2678** [comment]
It's annoying that Nest limit's us to 5% precision when logging RH.

Tracking RH from the logs for the past 6 hours doesn't reveal any noticeable difference when running the fan for 3 minutes after each cooling cycle:

![image](https://user-images.githubusercontent.com/8219743/28177314-6af791e8-67c8-11e7-8476-7d13759ff327.png)

Tracking RH manually in the Nest app (which reports indoor humidity down to 1% increments) the RH was 43% at the end of the last cooling cycle and stayed at 43% after running the fan an additional 3 minutes.

The WebCore piston I used to simulate Airwave, with your suggestion incorporated to only run under an RH % threshold is below:
![image](https://user-images.githubusercontent.com/8219743/28177085-aee3dd72-67c7-11e7-942e-029d9db5ecba.png)

Happy to hear any feedback you might have.

**Comment by tonesto7** [comment]
I’ve come to the realization that I think the 5% increments are hardware related because my third GEN thermostat increments and 1%

**Comment by johnny2678** [comment]
Interesting... I've got 2nd gen Nests.  Smartthings only logs in 5% increments but the official app reports 1%:

![image](https://user-images.githubusercontent.com/8219743/28191969-671c04a8-6801-11e7-8382-a3c88096d542.png)

**Comment by tonesto7** [comment]
It's not that SmartThings only reports in 5% increments it's actually the Nest API that returns that value.  Smartthings i believe would accept a humidity value 11.3% if the value is set that way

**Comment by johnny2678** [comment]
Sorry, correct - the Nest API is only reporting in 5% increments but obviously the sensor is capable of more precise values.

I suspect the temp sensor in the Nest is capable of detecting fractional temp changes too, but temp changes reported by the API are always rounded to the nearest whole #.
